src/storage/sqlite/xerial/src/main/java/org/locationtech/geogig/storage/sqlite/XerialObjectDatabaseV2.java

```diff
@@ -352,10 +353,12 @@ public void putAll(Iterator<? extends RevObject> objects, final BulkOpListener l
         checkWritable();
 
         // System.err.println("put allllllll");
-        txHandler.startTransaction();
-        while (objects.hasNext()) {
+        //txHandler.startTransaction();
```

Comment:
Still can't wrap the SQL edits here into a single transaction. Concurrent reads still happen during a putAll that aren't able to get at the nodes *put* in, even with **read_uncommitted** set. It's not terrible as this way, it will batch up to 50 RevObjects at a time into 1 commit transaction. So for very large numbers of RevObjects, we may have a performance penalty, but that is better than a corrupt or unpredictable database.